multipy/runtime/interpreter/plugin_registry.cpp

```diff
@@ -0,0 +1,34 @@
+#include "plugin_registry.h"
+
+#include <vector>
+
+namespace multipy {
+
+std::vector<Converter*>& getConverters() {
+  static std::vector<Converter*> converters;
+  return converters;
+}
+
+void registerConverter(Converter* c) {
+  getConverters().emplace_back(c);
+}
+
+at::IValue toTypeInferredIValue(py::handle input) {
+  for (auto c : getConverters()) {
+    auto out = c->toTypeInferredIValue(input);
```

Comment:
What would happen if we register more than one converter? Is it well defined/easy to figure out which converter would act first in general/multi-threaded situations? 

It may be better to make an OrderedConverter class which has this behavior similar to the OrderedImporter and Importer in package. I feel like this form of registry isn't correct here.

Reply:
gotcha, if the behavior is well defined by the order of the executables, I think it should be fine. I do feel like relying on the build system for debugging is probably not the right move as I feel like it's not really obvious to a lot of engineers. Though I can't really think of a cleaner way to do this without a somewhat annoying refactor.

Though, I'm not sure about the overlap situation, is that enforced or are we just hoping people are responsible about not having overlap? The main reason I'm asking is that I imagine, there will likely be a default converter for common types correct?

We could either throw an error during overlap, or add some debugging tools to figure out where they come from.

Reply:
The load order will be controlled by us since it's dynamically loaded at runtime and not via the compiler/system

The overlap I'm not too concerned about -- this pattern of trying them sequentially and falling back is the same way that the pybind type conversions work

https://pybind11.readthedocs.io/en/stable/advanced/cast/custom.html